src/vs/workbench/contrib/welcomeDialog/browser/welcomeDialog.contribution.ts

```diff
@@ -74,22 +75,29 @@ class WelcomeDialogContribution extends Disposable implements IWorkbenchContribu
 				const codeEditor = this.codeEditorService.getActiveCodeEditor();
 				if (codeEditor?.hasModel()) {
 
-					const detailsRenderer = new GettingStartedDetailsRenderer(fileService, notificationService, extensionService, languageService);
+					const scheduler = this._register(new RunOnceScheduler(() => {
+						const detailsRenderer = new GettingStartedDetailsRenderer(fileService, notificationService, extensionService, languageService);
 
-					const welcomeWidget = new WelcomeWidget(
-						codeEditor,
-						instantiationService,
-						commandService,
-						telemetryService,
-						openerService,
-						webviewService,
-						detailsRenderer);
+						const welcomeWidget = new WelcomeWidget(
+							codeEditor,
+							instantiationService,
+							commandService,
+							telemetryService,
+							openerService,
+							webviewService,
+							detailsRenderer);
 
-					welcomeWidget.render(welcomeDialog.title,
-						welcomeDialog.message,
-						welcomeDialog.buttonText,
-						welcomeDialog.buttonCommand,
-						welcomeDialog.media);
+						welcomeWidget.render(welcomeDialog.title,
+							welcomeDialog.message,
+							welcomeDialog.buttonText,
+							welcomeDialog.buttonCommand,
+							welcomeDialog.media);
+
+					}, 4000));
+
+					codeEditor.onDidChangeModelContent((_) => {
+						scheduler.schedule();
```

Comment:
Does this cause a re-schedule every time the model content changes?

Reply:
yes. fixing it currently by checking for already scheduled.